scala/private/rule_impls.bzl

```diff
@@ -318,10 +318,10 @@ StatsfileOutput: {statsfile_output}
     )
 
 def _interim_java_provider_for_java_compilation(scala_output):
```

Comment:
I think I need to read the code a bit more in depth but from what I remember this won't work because this way we won't have the java code present at runtime. I think the whole "interim" approach is now wrong and was needed before.
I might be mistaken but my hunch is that a bit more of a refactor is needed to have the scala compilation propagate out the JavaInfo of the java (or a different refactoring)

Reply:
What is the reason why `runtime_jars` was previously empty? 

IIUC this provider is needed to compile Java sources with scala dependencies. Does scala also need this provider?

Reply:
Yes, exactly. This only works because it's neverlink, otherwise we need to pass in the full jar as well (output_jar=full_jar, compile_jar=ijar) to avoid having an ijar at runtime.